mappings/UofM.py

```diff
@@ -75,6 +70,49 @@ def formatIdentifiers(self):
                     return newISBNList
                 
         return self.record.identifiers
+    
+    def formatSubjects(self):
+        subjectList = []
+
+        if '|' in self.record.subjects:
+            subjectList = self.record.subjects.split('|')
+            newSubjectList = [f'{subject}||' for subject in subjectList] 
+            return newSubjectList
+        else:
+            subjectList.append(f'{self.record.subjects}||')
+            return subjectList
+    
+    def formatRights(self):
+        # The pipe delimiter is to separate the Rights table attributes into this format:
+        # source|license|reason|statement|date 
+        # which makes it easy to place the right data into the columns when clustered
+
+        rightsElements = self.record.rights.split('|') if self.record.rights else [''] * 5
+        rightsStatus = rightsElements[0]
+
+        if rightsStatus == 'in copyright':
+            licenseMeta = 'in_copyright'
+            statementMeta = 'In Copyright'
+
+            rightsObject = 'UofM|{}||{}|'.format(
+                licenseMeta,
+                statementMeta
+            ) 
+
+            return rightsObject
+        
+        if rightsStatus == 'public domain':
+            licenseMeta = 'public_domain'
+            statementMeta = 'Public Domain'
+
+            rightsObject = 'UofM|{}||{}|'.format(
+                licenseMeta,
+                statementMeta
+            ) 
+
+            return rightsObject
+
+
```

Comment:
Let's return None here if the right status is neither public domain nor in copyright?